### PR TITLE
View/states policies

### LINKED ISSUE #12: Render data on Policies page and styling

### PR DESCRIPTION
**Pull request by catlynbowles**
** Resolves #12 **

** Describe the functionality to be tested: **

(Bullet list feature/functionality)
- Add info strips to state policies with state name and current legal status
- Add user ability to search
- Allow policies to be filtered by legal status
- Add search bar

** Giphy of how you feel submitting this PR: **

https://giphy.com/clips/stephanieweber-september-apple-picking-hello-LYqmeMhPTH1HSh6h6w

### CODE REVIEW

File: src/Views/StatePoliciesView/LegalKey/LegalKey.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+
+const LegalKey = ({filterByLegality, clearSearch}) => {
+  return (
+    <div className='legal-key'>
+      <h2 className='clear' style={{backgroundColor: 'black'}} onClick={() => clearSearch()}>Clear Search</h2>
+      <h2 className='legal' onClick={() => filterByLegality('0')}>LEGAL</h2>
+      <h2 className='at-risk' onClick={() => filterByLegality('1')}>AT RISK</h2>
+      <h2 className='illegal' onClick={() => filterByLegality('2')}>ILLEGAL</h2>
```
Comment (irmakerdem) [inline src/Views/StatePoliciesView/LegalKey/LegalKey.js:9]: Nice use of enums!

**Review by Rallen13 — APPROVED** [review]
Thanks for pushing this up! This will really help with completing my view

File: src/Views/StatePoliciesView/StatePoliciesView.js [hunk 2]
```diff
@@ -1,44 +1,88 @@
...
+  const [filteredResults, setFilteredResults] = useState([])
+  const inputRef = useRef(null)
   
-  return <h1>policies</h1>;
+  const handleSearchChange = (value) => {
+    setSearchInput(value);
+  };
+  
+  useEffect(() => {
+    if (searchInput !== '') {
```
Comment (lkarpins) [inline src/Views/StatePoliciesView/StatePoliciesView.js:39]: Great work with the search, Catlyn! 🔎

File: src/Views/StatePoliciesView/StatePoliciesView.js [hunk 3]
```diff
@@ -1,44 +1,88 @@
-import React from "react";
+import React, {useEffect, useState, useRef } from "react";
 import { useQuery, gql } from '@apollo/client'
+import PolicyCard from "./PolicyCard/PolicyCard";
+import LegalKey from "./LegalKey/LegalKey";
```
Comment (lkarpins) [inline src/Views/StatePoliciesView/StatePoliciesView.js:4]: Great work using graphQL!

**Review by lkarpins — APPROVED** [review]

File: src/Views/StatePoliciesView/StatePoliciesView.js [hunk 4]
```diff
@@ -1,44 +1,88 @@
...
+    <section className='policy-body'>
+      <h1 className='policy-header'>Check state abortion status</h1>
+      <div className="content-body">
+        <SearchBar inputRef={inputRef} handleSearchChange={handleSearchChange}/>
+        <div className='key-container'>
+          <img src={fight} height='200' alt='fight for your right megaphone'></img>
+          <LegalKey filterByLegality={filterByLegality} clearSearch={clearSearch}/>
+          <img src={female} height='200' alt='future is female with fists'></img>
+        </div>
+          {searchInput && legalLevel ? generatePolicyCards(filteredResults.filter(res => legalLevel.includes(res))) :
```
Comment (irmakerdem) [inline src/Views/StatePoliciesView/StatePoliciesView.js:79]: So many ternaries! ❤️

**Review by irmakerdem — APPROVED** [review]
Awesome job, Catlyn!

**Pull request merged**